File: learnDsaSeries/python/selection.py
class selection:
    def __init__(self,arr):
        self.arr=arr


    #sort the array 
    def sort(self):
        n=len(self.arr)
        for i in range(n):
            min=self.arr[i]
            p=i
            for j in range(i+1,n):
                if(min<self.arr[j]):
                    min=self.arr[j]
                    p=j

            self.arr[i],self.arr[p]=self.arr[p],self.arr[i]
        
        return self.arr
    
#create an instance of class 
lst=[23,1,345,43,2,67,3,899,34,68,49,89]
obj=selection(lst)
print(obj.sort())

File: jsWithSir/selection.js
class selection{
    constructor(data){
        this.data=data
    }

    // sort using selection sort
    sort(){
        for(let i=0;i<this.data.length;i++){
            let min=this.data[i]
            let p=i
            for(let j=i+1;j<this.data.length;j++){
                if(min>this.data[j]){
                    min=this.data[j]
                    p=j
                }
            }
            let temp=this.data[i]
            this.data[i]=this.data[p]
            this.data[p]=temp
        }
        return this.data
    }
}


let ar=[12,1,434,56,23,57,89,34,567,24,676]
obj=new selection(ar)
console.log(obj.sort())